server/service/src/sync/README.md

```diff
@@ -1,3 +1,99 @@
+
+SyncTranslator implementations are instantiated in an array. Using visitor pattern, this array of translators is used to check each translator against each sync buffer record when processing pulled records, and the resulting database operations are executed, we call the latter step integration.
+
+The same array of translators is used to process ChangeLogs when pushing records to `central server`.
+
+## ChangeLog
+
+[ChangeLog](https://github.com/msupply-foundation/open-msupply/blob/bc83acbb3cd51fe3375ac01135c6eb880a793936/server/repository/src/db_diesel/changelog/changelog.rs#L11) table is used to keep track of records that need to be pushed to central server. Database triggers will insert updated/deleted record’s table name, uuid and auto increment sequence number in ChangeLog table. During push operation we use a cursor to ChangeLog’s sequence number (RemoteSyncPushCursor) to figure out which records need to be pushed.
+
+To avoid re-pushing remote/patient and transfer records that have just arrived via sync, we use is_sync_updated field on that record, which is toggled to “true” during sync operation and toggled to “false” when record was edited outside of sync. 
+
+Also after initialisation, we update RemoteSyncPushCursor to point to the end of the ChangeLog sequence, to avoid resyncing records that have just arrived.
+
+## Integrations
+
+Translations will return an array of boxed [Upsert or Delete](https://github.com/msupply-foundation/open-msupply/blob/bc83acbb3cd51fe3375ac01135c6eb880a793936/server/repository/src/lib.rs#L49) traits, which are stored in IntegrationOperation enum. These traits are implemented by every sync data type, in the repository layer. 
+To ensure that the whole record in its entirety (invoice and all it’s lines) is available after synchronisation, we integrate everything in one big transaction. However this does slow down the initialisation phase, and thus transaction is skipped during initialisation, this is safe because the app is not available during the initialisation stage.
+
+
+## Example of Adding translation
+
+Please see below commit for an example of adding a new table and relative translations (it’s only an example and assumes that central server can already handle this record).
+
+[Add asset table](https://github.com/msupply-foundation/open-msupply/commit/dfa5b8e4a4d0a906b0c19f3ee00e194344834508)
+
+[Add translations](https://github.com/msupply-foundation/open-msupply/commit/a9b3d90a597f9670687f0fc26a5632310fe6b4c9)
+
+## Diagrams
+
+From [TMF internal docs](https://app.diagrams.net/#G1vVZQF_o0JiABBs1jpI0TneFfRonVqySD)
```

Comment:
this is nit picky! there's a spelling error in the linked diagram `syncrhoniser`

Reply:
Ouch, i'll update it in the downstream PR, since diagram has changed, thanks

Reply:
62d85d5bf8c6c37ad41a7476f9d10d4b7a16eec5, also moved diagram to code base, with Draw.io extension it's really cool